chain/src/txhashset/utxo_view.rs

```diff
@@ -21,12 +21,14 @@ use crate::core::global;
 use crate::core::ser::PMMRIndexHashable;
 use crate::error::{Error, ErrorKind};
 use crate::store::Batch;
+use crate::util::secp::pedersen::RangeProof;
 use grin_store::pmmr::PMMRBackend;
 
 /// Readonly view of the UTXO set (based on output MMR).
 pub struct UTXOView<'a> {
 	output_pmmr: ReadonlyPMMR<'a, Output, PMMRBackend<Output>>,
 	header_pmmr: ReadonlyPMMR<'a, BlockHeader, PMMRBackend<BlockHeader>>,
+	rproof_pmmr: ReadonlyPMMR<'a, RangeProof, PMMRBackend<RangeProof>>,
```

Comment:
Hmm. This makes the `utxo_view` even more similar to the `txhashset` itself. I _suspect_ we're going to want to get rid of the `utxo_view` at some point and just rely on the `txhashset` itself for more use cases like this.

The limiting factor is the writeable nature of the `txhashset` and the fact we need to take a write lock on it even if just reading from it.

I have a branch locally messing around with an attempt to `try_clone` the MMR backend itself. The idea being we could clone the txhashset to get a readonly view of the files, relying on the internal mmap implementation to give a consistent view of the data.

Reply:
That'd make sense and could significantly reduce the code complexity.